pkg/flags/github.go

```diff
@@ -56,4 +61,87 @@ func (g *GitHubFlags) Register(set *cli.FlagSet) {
 		Example: "repository-name",
 		Usage:   "The GitHub repository name.",
 	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-app-id",
+		EnvVar: "GITHUB_APP_ID",
+		Target: &g.FlagGitHubAppID,
+		Usage:  "The ID of GitHub App to use for requesting tokens to make GitHub API calls.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-app-installation-id",
+		EnvVar: "GITHUB_APP_INSTALLATION_ID",
+		Target: &g.FlagGitHubAppInstallationID,
+		Usage:  "The Installation ID of GitHub App to use for requesting tokens to make GitHub API calls.",
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:   "github-app-private-key-resource-name",
+		EnvVar: "GITHUB_APP_PRIVATE_KEY_RESOURCE_NAME",
+		Target: &g.FlagGitHubAppPrivateKeyResourceName,
+		Usage:  "The resource name of the private key to use with the GitHub App.",
+	})
+
+	set.AfterParse(func(merr error) error {
+		if g.FlagGitHubToken == "" && g.FlagGitHubAppID == "" {
+			merr = errors.Join(merr, fmt.Errorf("one of github token or github app id are required"))
+		}
+		if g.FlagGitHubToken != "" && g.FlagGitHubAppID != "" {
+			merr = errors.Join(merr, fmt.Errorf("only one of github token or github app id are allowed"))
+		}
+		if g.FlagGitHubAppID != "" && g.FlagGitHubAppInstallationID == "" {
+			merr = errors.Join(merr, fmt.Errorf("a github app installation id is required when using a github app id"))
+		}
+		if g.FlagGitHubAppID != "" && g.FlagGitHubAppPrivateKeyResourceName == "" {
+			merr = errors.Join(merr, fmt.Errorf("a github app private key resource name is required when using a github app id"))
+		}
+		return merr
+	})
+}
+
+func (g *GitHubFlags) TokenSource(ctx context.Context, permissions map[string]string) (githubauth.TokenSource, error) {
+	if g.FlagGitHubToken != "" {
+		githubTokenSource, err := githubauth.NewStaticTokenSource(g.FlagGitHubToken)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create github static token source: %w", err)
+		}
+		return githubTokenSource, nil
+	} else {
+		sm, err := secretmanager.NewClient(ctx)
```

Comment:
I think I would rather see this read from an env var or from a file, this way we could support Cloud Run (even as a job) in the future. Im not sure we need to couple to secrets manager here?

Additionally, this would let users test locally and other scenarios Im probably not thinking of.

Reply:
I said that earlier and you said there was no Cloud Run 😛 . But I agree: https://github.com/abcxyz/guardian/pull/291/files#r1493006723

Reply:
I changed it to read the private key from an environment variable. And yeah Seth did suggest that earlier lol.